doc/contributing.rst

```diff
@@ -21,6 +21,37 @@ of the users who use the package.
 page as the maintainers about changes or enhancements before too much
 coding is done saves everyone time and effort!
 
+Installing developer version
+----------------------------
+
+The standard installation procedure can be found on
+:ref:`install_python_and_mne_python` but if you want to install a
+developer version directly from the source code in order to contribute,
+you can follow those steps:
+
+  .. code-block:: console
+
+     $ git clone git@github.com:mne-tools/mne-python.git
+     $ cd mne-python
+     $ conda env create -f environment.yml
+     $ conda activate mne
+     $ pip install -e .
+
+As optional steps, you can also add:
+
+  .. code-block:: console
+
+     $ conda install sphinx
+     $ pip install sphinx_fontawesome sphinx_bootstrap_theme "https://api.github.com/repos/sphinx-gallery/sphinx-gallery/zipball/master"
```

Comment:
we really need sphinx-gallery master?

Reply:
I integrated suggestions from  [#5767 (comment)](#5767) maybe it's a more open question. I can update depending on feedback.